src/Core.js

```diff
@@ -71,6 +72,35 @@ export const login = (conf, callback, scope = 'info') => new Promise(((resolve,
   doLogin(url);
 }));
 
+export const apiLogin = async (conf, username, password, scope = 'info') => {
+  const {
+    resource, realm, credentials, 'auth-server-url': authServerUrl,
+  } = conf;
+
+  const url = `${getRealmURL(realm, authServerUrl)}/protocol/openid-connect/token`;
+  const method = POST;
+  const body = qs.stringify({
+    grant_type: 'password',
+    username,
+    password,
+    client_id: encodeURIComponent(resource),
+    client_secret: credentials ? credentials.secret : undefined,
+    scope,
+  });
+  const options = { headers: basicHeaders, method, body };
+
+  const fullResponse = await fetch(url, options);
+  const jsonResponse = await fullResponse.json();
+
+  if (fullResponse.status === 200) {
+    await TokenStorage.saveConfiguration(conf);
```

Comment:
I like the save conf choice. Maybe we can remove the conf argument from the others methods. Every time a plugin method need the conf information it'll read from the AsyncStorage. If that approach won't add too much latency, maybe can be an option.

Reply:
This is a great idea. and no, the latency will be exactly the same, since the user must read the configuration from the storage by himself, before passing it to the plugin! I'll implement the change